src/expr/src/scalar/func.rs

```diff
@@ -1205,6 +1205,56 @@ fn map_contains_key<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
     }
 }
 
+fn map_contains_all_keys<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+    let map = a.unwrap_map();
+    let keys = b.unwrap_array();
+
+    keys.elements()
+        .iter()
+        .all(|key| {
+            map.iter().any(|(k, _v)| {
+                // Maps cannot contain NULL keys.
+                if key.is_null() {
+                    false
+                } else {
+                    k == key.unwrap_str()
+                }
+            })
+        })
+        .into()
+}
+
+fn map_contains_any_keys<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+    let map = a.unwrap_map();
+    let keys = b.unwrap_array();
+
+    keys.elements()
+        .iter()
+        .any(|key| {
+            map.iter().any(|(k, _v)| {
+                // Maps cannot contain NULL keys.
+                if key.is_null() {
+                    false
```

Comment:
Ohh, my apologies! I got confused between the keys in the map and the search keys. From a performance perspective what you had before was better! 🙈 Or, alternatively, something like this if you want to do it all in one pass:

```
keys.elements().iter().any(|key| {
    !key.is_null() && map.iter().any(|k, _v| k == key.unwrap_str())
})
```

Just to avoid asking the question "is the needle NULL?" for every key in the haystack.